syncer/dml.go

```diff
@@ -443,20 +444,42 @@ func findColumns(columns []*column, indexColumns map[string][]string) map[string
 
 func genKeyList(table string, columns []*column, dataSeq []interface{}) string {
 	var buf strings.Builder
-	buf.WriteString(table)
 	for i, data := range dataSeq {
+		if data == nil {
+			log.L().Debug("ignore null value", zap.String("column", columns[i].name), zap.String("table", table))
+			continue // ignore `null` value.
+		}
 		// for key, I think no need to add the `,` separator.
 		buf.WriteString(columnValue(data, columns[i].unsigned, columns[i].tp))
 	}
+	if buf.Len() == 0 {
+		log.L().Debug("all value are nil, no key generated", zap.String("table", table))
+		return "" // all values are `null`.
+	}
+
+	buf.WriteString(table)
```

Comment:
why write table at the end?

Reply:
if all columns are `null`, the return `""` (without `table`).